server/routes/api/dashboard.js

```diff
@@ -0,0 +1,93 @@
+const express = require('express');
+const router = express.Router();
+let {
+    Course
+} = require('../../dbSchemas/attendanceSchema');
+
+
+/**GETS all the courses registered by a student
+ * ==========================================
+ * Example api call body:
+ * {
+	"studentId": "5f984a44da9eb32ba01d31dd"
+    }
+ */
+router.get('/getCoursesByStudent', (req, res) => {
+    let studentID = req.body.studentId;
+
+    Course.find({
+        registered_students: {
+            $in: [studentID]
+        }
+    }, function (err, courses) {
+        if (err || courses == null) {
+            console.log(err);
+            return res.status(500).send(err);
+        }
+        res.status(200).send(courses);
+    });
+});
+
+/**GETS all the courses created by a professor
+ * ==========================================
+ * Example api call body:
+ * {
+	"studentId": "5f98967d62a8bb73a0f8c046"
+    }
+ */
+router.get('/getCoursesCreatedByProfessor', (req, res) => {
+    let professorID = req.body.professorId;
+
+    Course.find({
+        professor: professorID
+    }, function (err, courses) {
+        if (err || courses == null) {
+            console.log(err);
+            return res.status(500).send(err);
+        }
+        res.status(200).send(courses);
+    });
+});
+
+/** Registers a student to his course
+ *  We give a student ID
+ *  Returns course data (name, prof name, if it's mandatory)
+ */
+router.post('/registerForCourse', (req, res) => {
+    let studentID = req.body.studentID;
+    let courseID = req.body.courseID;
+    console.log("register for course");
+
+    Course.findOne({
+        _id: courseID
+    }, function (err, course) {
+        if (err || course == null) {
+            console.log(err);
+            return res.status(500).send(err);
+        }
+
+        // push new student into array
+        course.registered_students.push(studentID);
```

Comment:
We should add a check to make sure the student is not already registered for the course. You can check if the student's id is in the array using the `indexOf()` function. If they are already registered it should return an error (can you return a different status code, maybe 520, so the frontend can give a specific message for this case).

Reply:
Done!